### PR TITLE
Add datastore compatability

### LINKED ISSUE #18: Unable to install Kar feature nexus-repository-terraform
Hi,

Could use some advice please on a nexus-repository-terraform startup issue. 

* **What are you trying to do?**
* Trying to get nexus to start with the terraform provider plugin active.

* **What feature or behavior is this required for?**
* terraform provider proxy.

* **How could we solve this issue? (Not knowing is okay!)**
* Dunno

* **Anything else?**
* errors
* 4 14:01:57,188+0000 ERROR [features-2-thread-1] *SYSTEM java.lang.Throwable - org.apache.felix.resolver.reason.ReasonException: Unable to resolve root: missing requirement [root] osgi.identity; osgi.identity=nexus-repository-terraform; type=karaf.feature; version="[0.0.4,0.0.4]"; filter:="(&(osgi.identity=nexus-repository-terraform)(type=karaf.feature)(version>=0.0.4)(version<=0.0.4))" [caused by: Unable to resolve nexus-repository-terraform/0.0.4: missing requirement [nexus-repository-terraform/0.0.4] osgi.identity; osgi.identity=org.sonatype.nexus.plugins.nexus-repository-terraform; type=osgi.bundle; version="[0.0.4,0.0.4]"; resolution:=mandatory [caused by: Unable to resolve org.sonatype.nexus.plugins.nexus-repository-terraform/0.0.4: missing requirement [org.sonatype.nexus.plugins.nexus-repository-terraform/0.0.4] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.sonatype.nexus.repository.attributes)(version>=3.38.0))" [caused by: Unable to resolve org.sonatype.nexus.repository [255](R 255.0): missing requirement [org.sonatype.nexus.repository [255](R 255.0)] osgi.wiring.package; (&(osgi.wiring.package=com.orientechnologies.common.concur)(version>=2.2.0))]]]
2022-08-24 14:01:57,191+0000 ERROR [features-2-thread-1] *SYSTEM java.lang.Throwable - Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve nexus-repository-terraform/0.0.4: missing requirement [nexus-repository-terraform/0.0.4] osgi.identity; osgi.identity=org.sonatype.nexus.plugins.nexus-repository-terraform; type=osgi.bundle; version="[0.0.4,0.0.4]"; resolution:=mandatory [caused by: Unable to resolve org.sonatype.nexus.plugins.nexus-repository-terraform/0.0.4: missing requirement [org.sonatype.nexus.plugins.nexus-repository-terraform/0.0.4] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.sonatype.nexus.repository.attributes)(version>=3.38.0))" [caused by: Unable to resolve org.sonatype.nexus.repository [255](R 255.0): missing requirement [org.sonatype.nexus.repository [255](R 255.0)] osgi.wiring.package; (&(osgi.wiring.package=com.orientechnologies.common.concur)(version>=2.2.0))]]
2022-08-24 14:01:57,192+0000 ERROR [features-2-thread-1] *SYSTEM java.lang.Throwable - Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve org.sonatype.nexus.plugins.nexus-repository-terraform/0.0.4: missing requirement [org.sonatype.nexus.plugins.nexus-repository-terraform/0.0.4] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.sonatype.nexus.repository.attributes)(version>=3.38.0))" [caused by: Unable to resolve org.sonatype.nexus.repository [255](R 255.0): missing requirement [org.sonatype.nexus.repository [255](R 255.0)] osgi.wiring.package; (&(osgi.wiring.package=com.orientechnologies.common.concur)(version>=2.2.0))]
2022-08-24 14:01:57,196+0000 WARN [fileinstall-/opt/nexus/nexus-3.38.1-01/deploy] *SYSTEM org.apache.karaf.kar.internal.KarServiceImpl - Unable to install Kar feature nexus-repository-terraform/0.0.4
org.apache.felix.resolver.reason.ReasonException: Unable to resolve root: missing requirement [root] osgi.identity; osgi.identity=nexus-repository-terraform; type=karaf.feature; version="[0.0.4,0.0.4]"; filter:="(&(osgi.identity=nexus-repository-terraform)(type=karaf.feature)(version>=0.0.4)(version<=0.0.4))" [caused by: Unable to resolve nexus-repository-terraform/0.0.4: missing requirement [nexus-repository-terraform/0.0.4] osgi.identity; osgi.identity=org.sonatype.nexus.plugins.nexus-repository-terraform; type=osgi.bundle; version="[0.0.4,0.0.4]"; resolution:=mandatory [caused by: Unable to resolve org.sonatype.nexus.plugins.nexus-repository-terraform/0.0.4: missing requirement [org.sonatype.nexus.plugins.nexus-repository-terraform/0.0.4] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.sonatype.nexus.repository.attributes)(version>=3.38.0))" [caused by: Unable to resolve org.sonatype.nexus.repository [255](R 255.0): missing requirement [org.sonatype.nexus.repository [255](R 255.0)] osgi.wiring.package; (&(osgi.wiring.package=com.orientechnologies.common.concur)(version>=2.2.0))]]]
at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341)
at org.apache.felix.resolver.ResolverImpl.doResolve(ResolverImpl.java:433)
at org.apache.felix.resolver.ResolverImpl.resolve(ResolverImpl.java:420)
at org.apache.felix.resolver.ResolverImpl.resolve(ResolverImpl.java:374)
at org.apache.karaf.features.internal.region.SubsystemResolver.resolve(SubsystemResolver.java:257)
at org.apache.karaf.features.internal.service.Deployer.deploy(Deployer.java:399)
at org.apache.karaf.features.internal.service.FeaturesServiceImpl.doProvision(FeaturesServiceImpl.java:1069)
at org.apache.karaf.features.internal.service.FeaturesServiceImpl.lambda$doProvisionInThread$13(FeaturesServiceImpl.java:1004)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:750)
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve nexus-repository-terraform/0.0.4: missing requirement [nexus-repository-terraform/0.0.4] osgi.identity; osgi.identity=org.sonatype.nexus.plugins.nexus-repository-terraform; type=osgi.bundle; version="[0.0.4,0.0.4]"; resolution:=mandatory [caused by: Unable to resolve org.sonatype.nexus.plugins.nexus-repository-terraform/0.0.4: missing requirement [org.sonatype.nexus.plugins.nexus-repository-terraform/0.0.4] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.sonatype.nexus.repository.attributes)(version>=3.38.0))" [caused by: Unable to resolve org.sonatype.nexus.repository [255](R 255.0): missing requirement [org.sonatype.nexus.repository [255](R 255.0)] osgi.wiring.package; (&(osgi.wiring.package=com.orientechnologies.common.concur)(version>=2.2.0))]]
at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341)
... 12 common frames omitted
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve org.sonatype.nexus.plugins.nexus-repository-terraform/0.0.4: missing requirement [org.sonatype.nexus.plugins.nexus-repository-terraform/0.0.4] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.sonatype.nexus.repository.attributes)(version>=3.38.0))" [caused by: Unable to resolve org.sonatype.nexus.repository [255](R 255.0): missing requirement [org.sonatype.nexus.repository [255](R 255.0)] osgi.wiring.package; (&(osgi.wiring.package=com.orientechnologies.common.concur)(version>=2.2.0))]
at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1341)
... 13 common frames omitted

Thank you,
-dkw

cc @DarthHater @bhamail

### PR DESCRIPTION
**Pull request by davejab**
Adding datastore compatibility.

This pull request makes the following changes:
* TODO

It relates to the following issue #s:
* Fixes #18

cc @DarthHater @bhamail

### CODE REVIEW

File: nexus-repository-terraform/src/main/java/org/sonatype/nexus/plugins/terraform/internal/matcher/ProviderVersionsMatcher.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.plugins.terraform.internal.matcher;
+
+import org.sonatype.nexus.plugins.terraform.internal.AssetKind;
+
+import static org.sonatype.nexus.plugins.terraform.internal.util.TerraformPathUtils.PROVIDERS_PATH;
+
+/**
+ * @since 0.0.6
```
[the remark this answers is not in the document]
Comment (davejab) [reply nexus-repository-terraform/src/main/java/org/sonatype/nexus/plugins/terraform/internal/matcher/ProviderVersionsMatcher.java:20]: @sonatype-lift ignore

File: nexus-repository-terraform/src/main/java/org/sonatype/nexus/plugins/terraform/internal/Attributes/TerraformAttributes.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.plugins.terraform.internal.Attributes;
+
+import org.sonatype.nexus.plugins.terraform.internal.AssetKind;
+import org.sonatype.nexus.repository.view.matchers.token.TokenMatcher;
+
+/**
+ * @since 0.0.6
```
[the remark this answers is not in the document]
Comment (davejab) [reply nexus-repository-terraform/src/main/java/org/sonatype/nexus/plugins/terraform/internal/Attributes/TerraformAttributes.java:19]: @sonatype-lift ignore

File: nexus-repository-terraform/src/main/java/org/sonatype/nexus/plugins/terraform/internal/Attributes/TerraformArchiveAttributes.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.plugins.terraform.internal.Attributes;
+
+/**
+ * @since 0.0.6
```
[the remark this answers is not in the document]
Comment (davejab) [reply nexus-repository-terraform/src/main/java/org/sonatype/nexus/plugins/terraform/internal/Attributes/TerraformArchiveAttributes.java:16]: @sonatype-lift ignore

**Comment by damnski** [comment]
Hello @davejab,

I hope all is well.  Can you please advise what the status is?  Thanks for volunteering your time.

Best,
-dkw

**Comment by davejab** [comment]
@damnski apologies for the delay, I have been terribly busy over the past month. I believe this is mostly working I am just missing tests for the datastore access. I will try set aside some time in the near future to get this completed.

**Comment by damnski** [comment]
Thank you @davejab !  How can I buy you a beer or coffee or otherwise thank you?

**Pull request merged**